### PR TITLE
Update quick start docs

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Part of docs revamp #7666

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -20,7 +20,6 @@ PyTorch Lightning
 
    starter/new-project
    starter/converting
-   starter/rapid_prototyping_templates
```
Comment (awaelchli) [inline docs/source/index.rst:23]: Does this PR address #10268, #11082

Comment (awaelchli) [reply docs/source/index.rst:23]: ah no, it was just moved

File: docs/source/starter/new-project.rst [hunk 2]
```diff
@@ -12,35 +12,76 @@
...
 
-Here's a 3 minute conversion guide for PyTorch projects:
+Why Pytorch Lightning
+=====================
 
-.. raw:: html
+Well, if you need one place that can help you with everything you need to train deep learning models, you are at the right place. The team makes sure that all the latest techniques are already integrated and maintained
+periodically to make sure you don't need to make any code changes.
+
+We test the framework across multiple Python/PyTorch versions to ensure its reach. It's not just Keras for PyTorch but much more. Follow along with the guide below to find out why.
```
Comment (awaelchli) [inline docs/source/starter/new-project.rst:39]: The Keras comparison is out of place. What does it mean to be "more than Keras" anyway?

File: docs/source/starter/new-project.rst [hunk 3]
```diff
@@ -61,8 +102,33 @@ You could also use conda environments
     conda activate my_env
     pip install pytorch-lightning
 
+Install future patch release from the source
+
+.. code-block:: bash
+
+    pip install git+https://github.com/PytorchLightning/pytorch-lightning.git@release/1.5.x --upgrade
+
+Install nightly from the source (no guarantees)
```
Comment (awaelchli) [inline docs/source/starter/new-project.rst:111]: not convinced this should be part of the quick start guide, as 99% users will not want to quick start with nightly (why would they). This is distracting, as the user is has to make a choice.

File: docs/source/starter/new-project.rst [hunk 4]
```diff
@@ -61,8 +102,33 @@ You could also use conda environments
     conda activate my_env
     pip install pytorch-lightning
 
+Install future patch release from the source
```
Comment (awaelchli) [inline docs/source/starter/new-project.rst:105]: this is the quick start guide, meaning the user will install it fresh, so they anyway get the 1.5.x version. Including this upgrade process here is redundant.

**Review by awaelchli — CHANGES_REQUESTED** [review]
All the small formatting improvements and fixes to the code samples are great. I'm not a fan of the changes to the introduction. There are too bold statements about what Lightning is or what it does. 
For example, to say that "Lightning has all you need to do deep learning, you are in the right place" is the wrong approach to explain it. The way I like to distinguish Lightning is: "Lightning is not a framework, PyTorch is. Lightning is a philosophy." 

My view is that this page is meant to be a quick start guide, so it should make it quick for people to get started.

File: docs/source/starter/new-project.rst (hunk 2 above)
Comment (rohitgr7) [reply docs/source/starter/new-project.rst:39]: just wanted to specify that it's not just a wrapper around pytorch like keras where you plugin your model and do a `.fit`, but you have control over the loops and what happens inside each step.

File: docs/source/starter/new-project.rst [hunk 5]
```diff
@@ -12,35 +12,76 @@
...
 
-Organizing your code with PyTorch Lightning makes your code:
+What is Pytorch Lightning
+=========================
 
-* Keep all the flexibility (this is all pure PyTorch), but removes a ton of boilerplate
-* More readable by decoupling the research code from the engineering
-* Easier to reproduce
-* Less error-prone by automating most of the training loop and tricky engineering
-* Scalable to any hardware without changing your model
```
[the remark this answers is not in the document]
Comment (rohitgr7) [reply docs/source/starter/new-project.rst:27]: I'll put them back.

**Comment by rohitgr7** [comment]
> All the small formatting improvements and fixes to the code samples are great. I'm not a fan of the changes to the introduction. There are too bold statements about what Lightning is or what it does. For example, to say that "Lightning has all you need to do deep learning, you are in the right place" is the wrong approach to explain it. The way I like to distinguish Lightning is: "Lightning is not a framework, PyTorch is. Lightning is a philosophy."
> 
> My view is that this page is meant to be a quick start guide, so it should make it quick for people to get started.

@awaelchli Marc suggested putting in what is Lightning, and why someone wants to use it.. so that's why I added those 2 extra sections.

Do you think there should be 2 separate sections in the sidebar?

```
Introduction
Quick start
```
??

Then we can move it out of the quick start guide.

File: docs/source/starter/new-project.rst (hunk 3 above)
Comment (rohitgr7) [reply docs/source/starter/new-project.rst:111]: just wanted to introduce a section that tells all possible ways of installation. Pytorch mentions nightly too on their website: https://pytorch.org/

but you are right, maybe if we agree on the introduction page, this can be moved completely out of this one and we can just mention the installation guide for the published one.

File: docs/source/starter/installation.rst [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+************************
+Installation from Source
+************************
+
+Install future patch release from the source. Note that patch release contains only the bugfixes for the recent major release.
+
+.. code-block:: bash
+
+    pip install git+https://github.com/PytorchLightning/pytorch-lightning.git@release/1.5.x --upgrade
```
Comment (Borda) [inline docs/source/starter/installation.rst:47]:
```suggestion
    pip install https://github.com/PyTorchLightning/pytorch-lightning/archive/refs/heads/release/1.5.x.zip
```
git is heavy

**Review by Borda — APPROVED** [review]

File: docs/source/starter/new-project.rst (hunk 3 above)
Comment (Borda) [reply docs/source/starter/new-project.rst]: yeah but their nightly is not bare master but still need to tested and compiled which is not the same as you suggest user to use PL sourcecode :]

File: docs/source/starter/new-project.rst (hunk 4 above)
Comment (Borda) [reply docs/source/starter/new-project.rst]: what we can do is double branches, meaning the actual `release/1.5.x` will be called `release/stable` and when we start with v1.6, the rename `release/stable` back to `release/1.5.x`

File: docs/source/starter/installation.rst [hunk 7]
```diff
@@ -0,0 +1,55 @@
+.. _installation:
+
+############
+Installation
+############
+
+PyTorch Lightning is maintained and tested on Pytorch 3.7.0+ and PyTorch 1.7.0+.
```
Comment (tchaton) [inline docs/source/starter/installation.rst:7]: This should be a more exhaustive matrix, similar to the readme.

File: docs/source/starter/installation.rst [hunk 8]
```diff
@@ -0,0 +1,55 @@
+.. _installation:
+
+############
+Installation
```
Comment (tchaton) [inline docs/source/starter/installation.rst:4]: This isn't about installation. Maybe let's rename this Lightning Coverage

**Review by tchaton — APPROVED** [review]
LGTM !

File: docs/source/starter/installation.rst (hunk 7 above)
Comment (rohitgr7) [reply docs/source/starter/installation.rst:7]: I can improve it a little more, but adding that exhaustive matrix might be an overkill. I don't think users might be interested in what kind of tests we cover in the docs. Plus we need to make sure that we update this particular doc every time a small change is made in the ci config. As far as I can tell, the coverage on README is a little outdated too.

File: docs/source/starter/installation.rst (hunk 8 above)
Comment (rohitgr7) [reply docs/source/starter/installation.rst:4]: it's about installation. Lightning coverage is just a small part to indicate whether users' env will be supported with PL or not.

**Comment by rohitgr7** [comment]
requesting @awaelchli for re-review.

File: docs/source/starter/installation.rst (hunk 7 above)
Comment (carmocca) [reply docs/source/starter/installation.rst:11]: If this is already in the README, I would not copy it here.

We will forget about this eventually. My suggestion is to remove it or link to the README.

File: docs/source/starter/installation.rst [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+.. code-block:: bash
+
+    conda install pytorch-lightning -c conda-forge
+
+You can also use `Conda Environments <https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html>`_:
+
+.. code-block:: bash
+
+    conda activate my_env
+    conda install pytorch-lightning -c conda-forge
```
Comment (carmocca) [inline docs/source/starter/installation.rst:45]: What's the benefit of mentioning this? This is not lightning-specific right?

File: docs/source/starter/installation.rst [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+Install future patch release from the source. Note that patch release contains only the bugfixes for the recent major release.
+
+.. code-block:: bash
+
+    pip install https://github.com/PyTorchLightning/pytorch-lightning/archive/refs/heads/release/1.5.x.zip
+
+
+Install from the source. Note that it contains all the bugfixes and newly released features that
+are not published yet. Also newly added features might contain bugs so no gurantess.
```
Comment (carmocca) [inline docs/source/starter/installation.rst:59]: No guarantees reads a bit aggresive to me

```suggestion
are not published yet. This is the bleeding edge so use it at your own discretion.
```

File: docs/source/starter/new-project.rst [hunk 11]
```diff
@@ -12,57 +12,110 @@
...
+############
 
-**In this guide we'll show you how to organize your PyTorch code into Lightning in 2 steps.**
+*************************
+What is PyTorch Lightning
+*************************
 
-Organizing your code with PyTorch Lightning makes your code:
+PyTorch has all you need to train your models. But there's a lot more to deep learning more than just attaching some layers together. It provides you the APIs required to build models, datasets etc.
+but when it comes to actual training, there's a lot of boilerplate code involved that you need to write by yourself and if you need to scale your training/inferencing on multiple devices/machines there's another
```
Comment (carmocca) [inline docs/source/starter/new-project.rst:24]: Isn't this a new sentence?

```suggestion
When it comes to actual training, there's a lot of boilerplate code involved that you need to write by yourself. And if you need to scale your training/inferencing on multiple devices/machines there's another
```

File: docs/source/starter/new-project.rst [hunk 12]
```diff
@@ -12,57 +12,110 @@
...
+set of integrations you might need to do by yourself.
+
+So here PyTorch Lightning comes into the picture. All you need is some restructuring of your existing code and set certain flags and voila, you are done.
```
Comment (carmocca) [inline docs/source/starter/new-project.rst:27]:
```suggestion
PyTorch Lightning solves this for you. All you need is some restructuring of your existing code and set certain flags and voila, you are done.
```

File: docs/source/starter/new-project.rst [hunk 13]
```diff
@@ -12,57 +12,110 @@
...
+Now you can train your models on different accelerators like GPU/TPU/IPU, do distributed training accross multiple machines/nodes with no code change use state-of-the-art distributed training mechanisms
+and a lot more. Now all you need is to take care of your research code and leave the engineering to us.
+
+Code organization is the core of Lightning. It leaves core training and validation logic to you and automates the rest.
```
Comment (carmocca) [inline docs/source/starter/new-project.rst:31]:
```suggestion
Code organization is the core of Lightning. It leaves the research logic to you and automates the rest.
```

File: docs/source/starter/new-project.rst [hunk 14]
```diff
@@ -12,57 +12,110 @@
...
+*************************
 
-Organizing your code with PyTorch Lightning makes your code:
+PyTorch has all you need to train your models. But there's a lot more to deep learning more than just attaching some layers together. It provides you the APIs required to build models, datasets etc.
+but when it comes to actual training, there's a lot of boilerplate code involved that you need to write by yourself and if you need to scale your training/inferencing on multiple devices/machines there's another
+set of integrations you might need to do by yourself.
+
+So here PyTorch Lightning comes into the picture. All you need is some restructuring of your existing code and set certain flags and voila, you are done.
+Now you can train your models on different accelerators like GPU/TPU/IPU, do distributed training accross multiple machines/nodes with no code change use state-of-the-art distributed training mechanisms
+and a lot more. Now all you need is to take care of your research code and leave the engineering to us.
```
Comment (carmocca) [inline docs/source/starter/new-project.rst:29]:
```suggestion
and a lot more.
```

This is saying the same as the sentence below

File: docs/source/starter/new-project.rst [hunk 15]
```diff
@@ -12,57 +12,110 @@
...
 
-Here's a 3 minute conversion guide for PyTorch projects:
+* Researcher who want to focus on research without worrying about the engineering aspects of it
+* ML Engineers who want to built reproducible pipelines
+* Data Scientists who want to try out different models for their tasks and build-in ML techniques
+* Educators who seek to study and teach Deep Learning with PyTorch
 
-.. raw:: html
+The team makes sure that all the latest techniques are already integrated and maintained periodically to make sure you don't
+have to worry about any engineering stuff and just focus on research.
```
Comment (carmocca) [inline docs/source/starter/new-project.rst:55]: This is repeating ourselves again

```suggestion
The team makes sure that all the latest techniques are already integrated and well maintained.
```

File: docs/source/starter/new-project.rst [hunk 16]
```diff
@@ -12,57 +12,110 @@
...
+
+It is tested across multiple Python/PyTorch versions to ensure its reach.
```
Comment (carmocca) [inline docs/source/starter/new-project.rst:57]: I don't think this sentence fits here, there's the installation and readme for it.

```suggestion
```

**Review by carmocca — APPROVED** [review]
I think this PR could have been split down further.

File: docs/source/starter/installation.rst (hunk 9 above)
Comment (rohitgr7) [reply docs/source/starter/installation.rst:45]: I think it's fine. If someone is not familiar with conda at all, then it's just a useful reference.

File: docs/source/starter/installation.rst (hunk 7 above)
Comment (rohitgr7) [reply docs/source/starter/installation.rst]: linked the README.

File: docs/source/starter/installation.rst [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+Installation
+############
+
+--------------
+
+******************
+Lightning Coverage
+******************
+
+PyTorch Lightning is maintained and tested on different Pytorch and PyTorch versions.
```
Comment (awaelchli) [inline docs/source/starter/installation.rst:13]:
```suggestion
PyTorch Lightning is maintained and tested on different PyTorch versions.
```

Comment (rohitgr7) [reply docs/source/starter/installation.rst:13]: ah! should be Python.. my fingers have learnt `Pytorch` word. 😅

Comment (rohitgr7) [inline docs/source/starter/installation.rst:13]:
```suggestion
PyTorch Lightning is maintained and tested on different Python and PyTorch versions.
```

Comment (awaelchli) [reply docs/source/starter/installation.rst:13]: Please help my eyes are bleeding. 
All the people spelling PyTorch wrong...
